codec-http2/src/test/java/io/netty/handler/codec/http2/Http2CodecTest.java

```diff
@@ -0,0 +1,144 @@
+public class Http2CodecTest {
+
+    private EmbeddedChannel channel;
+
+    private Http2FrameWriter frameWriter;
+    private TestChannelInitializer testChannelInitializer;
+
+    @Before
+    public void setUp() {
+        frameWriter = spy(new VerifiableHttp2FrameWriter());
+        testChannelInitializer = new TestChannelInitializer();
+        Http2StreamChannelBootstrap bootstrap = new Http2StreamChannelBootstrap().handler(testChannelInitializer);
+        channel = new EmbeddedChannel();
+        channel.connect(new InetSocketAddress(0));
+        channel.pipeline().addLast(new Http2Codec(true, bootstrap, frameWriter));
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        channel.finishAndReleaseAll();
+    }
+
+    @Test
+    public void multipleOutboundStreams() {
+        Http2StreamChannelBootstrap b = new Http2StreamChannelBootstrap();
+        b.parentChannel(channel).handler(testChannelInitializer);
+
+        Channel channel1 = b.connect().channel();
+        assertTrue(channel1.isActive());
+        assertFalse(((AbstractHttp2StreamChannel) channel1).hasStreamId());
+        Channel channel2 = b.connect().channel();
+        assertTrue(channel2.isActive());
+        assertFalse(((AbstractHttp2StreamChannel) channel2).hasStreamId());
+
+        Http2Headers headers1 = new DefaultHttp2Headers();
+        Http2Headers headers2 = new DefaultHttp2Headers();
+        // Test that streams can be made active (headers sent) in different order than the corresponding channels
+        // have been created.
+        channel2.writeAndFlush(new DefaultHttp2HeadersFrame(headers2));
+        channel1.writeAndFlush(new DefaultHttp2HeadersFrame(headers1));
+
+        verify(frameWriter).writeHeaders(any(ChannelHandlerContext.class), eq(2), same(headers2), anyInt(), anyShort(),
+                                         eq(false), eq(0), eq(false), any(ChannelPromise.class));
+        verify(frameWriter).writeHeaders(any(ChannelHandlerContext.class), eq(4), same(headers1), anyInt(), anyShort(),
+                                         eq(false), eq(0), eq(false), any(ChannelPromise.class));
+
+        assertTrue(((AbstractHttp2StreamChannel) channel1).hasStreamId());
+        assertTrue(((AbstractHttp2StreamChannel) channel2).hasStreamId());
+
+        channel1.close();
+        channel2.close();
+    }
+
+    @Test
+    public void createOutboundStream() {
+        Http2StreamChannelBootstrap b = new Http2StreamChannelBootstrap();
+        ChannelFuture channelFuture = b.parentChannel(channel).handler(testChannelInitializer).connect();
+        assertTrue(channelFuture.isSuccess());
+        Channel childChannel = channelFuture.channel();
+        assertTrue(childChannel.isRegistered());
+        assertTrue(childChannel.isActive());
+
+        Http2Headers headers = new DefaultHttp2Headers();
+        childChannel.write(new DefaultHttp2HeadersFrame(headers));
+        ByteBuf data = Unpooled.buffer(100).writeZero(100);
```

Comment:
The method used by other tests is different and doesn't seem to run into the same issue. Would it be sufficient to read from the EmbeddedChannel's outbound queue and verify the messages you expect to receive instead of validating internal state?